### DISCUSSION TITLE
helm chart - change default dashboard - disable RSSFeedWidget

### DISCUSSION THREAD

**Discussion by salcinad** [netbox-community/netbox#18216] [Q&A]
I would like to disable RSSFeedWidget in Kubernetes deployment with helm chart. I was trying to use extraConfig and the DEFAULT_DASHBOARD part from [constants.py](https://github.com/netbox-community/netbox/blob/develop/netbox/extras/constants.py) as values, but ArgoCD is unable to render this.
```
        extraConfig:
          - values:
              DEFAULT_DASHBOARD: [
                {
                },
            ]
```

This is the Sync Error from ArgoCD which we get:
`one or more objects failed to apply, reason: error when patching "/dev/shm/1852142632": ConfigMap "netbox" is invalid: data[extra-%!s(int=0).yaml]: Invalid value: "extra-%!s(int=0).yaml": a valid config key must consist of alphanumeric characters, '-', '_' or '.' (e.g. 'key.name', or 'KEY_NAME', or 'key-name', regex used for validation is '[-._a-zA-Z0-9]+'),error when patching "/dev/shm/990010489": Deployment.apps "netbox" is invalid: spec.template.spec.containers[0].volumeMounts[4].name: Not found: "extra-config-%!s(int=0)",error when patching "/dev/shm/4005637113": Deployment.apps "netbox-worker" is invalid: spec.template.spec.containers[0].volumeMounts[4].name: Not found: "extra-config-%!s(int=0)",error when patching "/dev/shm/3336613721": CronJob.batch "netbox-housekeeping" is invalid: spec.jobTemplate.spec.template.spec.containers[0].volumeMounts[4].name: Not found: "extra-config-%!s(int=0)"`

We are using netbox chart version 5.0.0-beta.154 from [artifacthub](https://artifacthub.io/packages/helm/netbox/netbox). Without extraConfig deployment works fine, but as we are in an Airgap environment the RSSFeedWidget does not work and it takes a bit time to get Front Page and we guess it because of that.  There was also an Issue with similar problem [Issue 12384](https://github.com/netbox-community/netbox/issues/12384)

If there is any other way to disable RSSFeedWidget we would be glad to test it, currently we did not find and other way.

**Comment by LeoColomb** [comment]
https://github.com/netbox-community/netbox-chart/issues/464

**Reply by salcinad** [reply]
> [netbox-community/netbox-chart#464](https://github.com/netbox-community/netbox-chart/issues/464)

> Thanks for filing this issue, @alihoda. You can add any additional NetBox parameters using the values `extraConfig`. For instance:
> 
> ```yaml
> extraConfig:
>   - values:
>       ISOLATED_DEPLOYMENT: true
> ```
> 
> If you feel it's relevant to actually add this option explicitly in the chart, don't hesitate to open a pull request with this addition.
> 
> @salcinad Your use case is correct, you simply need to be careful about the `extraConfig` templating. The extra config is expected to be provided as a valid YAML. For example:
> 
> ```yaml
> extraConfig:
>   - values:
>       DEFAULT_DASHBOARD:
>         - {}
> ```

@LeoColomb In your comment you provided information how to do it with helm chart. We did today test with that but RSSFeedWidget is still there on Frontpage, and is in timeout because Air Gap, should it not be disabled?


```
$ previous_release="$(helm history -n netbox-dev netbox | tail -2 | head -1 | cut -f 1 -d' ')"
$ helm diff -n netbox-dev revision netbox "$previous_release"
netbox-dev, netbox, ConfigMap (v1) has changed:
..
+   extra-0.yaml: |-
+     ISOLATED_DEPLOYMENT: true
```

We still need to test with extraconfig for DEFAULT_DASHBOARD.
